docs/DafnyRef/Statements.md

```diff
@@ -1578,26 +1630,28 @@ is omitting the proof of the claim made by the lemma specification. As with the
 other body-less constructs above, the verifier is silently happy with a body-less
 `forall` statement, but the compiler will complain.
 
-## 20.15. Match Statement {#sec-match-statement}
-````grammar
-MatchStmt =
-  "match"
-  Expression(allowLemma: true, allowLambda: true)
-  ( "{" { CaseStmt } "}"
-  | { CaseStmt }
-  )
-
-CaseStmt = "case" ExtendedPattern "=>" { Stmt }
-````
+## 20.15. Match Statement ([grammar](#g-match-statement)) {#sec-match-statement}
 
-[ `ExtendedPattern` is defined in [Section 21.33](#sec-case-pattern).]
+Examples:
+<!-- %no-check -->
+```dafny
+match x case 1 => print x; case 2 => var y := x*x; print y; case _ => print "Other";
+match list { case Nil => {} case Cons(head,tail) => print head; }
```

Comment:
```suggestion

match list {
  case Nil => {}
  case Cons(head,tail) => print head;
}
match x
case 1 =>
  print x;
case 2 =>
  var y := x*x;
  print y;
case _ =>
  print "Other";
  // Any statement after is captured in this case.
```

Reply:
I was wanting to keep the examples compact, but we can go with the official formatting.